examples/medplum-photon-integration/src/bots/handle-prescription-event.ts

```diff
@@ -0,0 +1,306 @@
+ * @returns The updated prescription
+ */
+export async function handleUpdatePrescription(
+  body: PrescriptionDepletedEvent | PrescriptionExpiredEvent,
+  medplum: MedplumClient,
+  existingPrescription?: MedicationRequest
+): Promise<MedicationRequest> {
+  if (!existingPrescription) {
+    throw new Error('No prescription to update');
+  }
+
+  const updatedStatus = body.type === 'photon:prescription:depleted' ? 'completed' : 'stopped';
+  const identifiers = existingPrescription.identifier ?? [];
+  identifiers.push({ system: 'https://neutron.health/webhooks', value: body.id });
+  const op = existingPrescription.identifier ? 'replace' : 'add';
+
+  const id = body.data.externalId as string;
+  const ops: PatchOperation[] = [
+    { op: 'test', path: '/meta/versionId', value: existingPrescription.meta?.versionId },
+    { op: 'replace', path: '/status', value: updatedStatus },
+    { op, path: '/identifier', value: identifiers },
+  ];
+
+  try {
+    const result = await medplum.patchResource('MedicationRequest', id, ops);
+    return result;
+  } catch (err) {
+    throw new Error(normalizeErrorString(err));
+  }
+}
+
+/**
+ * Takes the data from the incoming webhook and uses it to create a prescription modeled as a FHIR MedicationRequest resource.
+ *
+ * @param event - The webhook event with the details needed to create the prescription
+ * @param medplum - Medplum Client to create the prescription in your project
+ * @param authToken - Photon auth token to authorize GraphQL queries
+ * @returns The created prescription as MedicationRequest resource
+ */
+export async function handleCreatePrescription(
+  event: PhotonEvent,
+  medplum: MedplumClient,
+  authToken: string
+): Promise<MedicationRequest> {
+  const data = event.data as PrescriptionCreatedData;
+  // Get the prescribing practitioner and the medication from Medplum
+  const prescriber = await getPrescriber(data, medplum, authToken);
+  let medication: MedicationKnowledge | undefined = await getMedicationKnowledge(data.medicationId, medplum, authToken);
+
+  if (!medication) {
+    medication = await createMedicationKnowledge(data, medplum, authToken);
+  }
+
+  const medicationRequest: MedicationRequest = {
+    resourceType: 'MedicationRequest',
+    status: 'active',
+    intent: 'order',
+    subject: { reference: `Patient/${data.patient.externalId}` },
+    identifier: [
+      { system: 'https://neutron.health/webhooks', value: event.id },
+      { system: 'https://neutron.health', value: data.id },
+    ],
+    dispenseRequest: {
+      quantity: {
+        value: data.dispenseQuantity,
+        unit: data.dispenseUnit,
+      },
+      numberOfRepeatsAllowed: data.refillsAllowed,
+      expectedSupplyDuration: { value: data.daysSupply, unit: 'days' },
+      validityPeriod: {
+        start: data.effectiveDate,
+        end: data.expirationDate,
+      },
+    },
+    substitution: { allowedBoolean: data.dispenseAsWritten },
```

Comment:
```suggestion
    substitution: { allowedBoolean: !data.dispenseAsWritten },
```

DAW means that substitutions are _not_ allowed